demux/demux_playlist.c

```diff
@@ -54,11 +66,34 @@ struct m_sub_options demux_playlist_conf = {
             {"lazy", DIR_LAZY},
             {"recursive", DIR_RECURSIVE},
             {"ignore", DIR_IGNORE})},
+        {"directory-filter-types",
+            OPT_STRINGLIST(directory_filter)},
+        {"directory-video-exts",
+            OPT_STRINGLIST(directory_vid_exts)},
+        {"directory-audio-exts",
+            OPT_STRINGLIST(directory_aud_exts)},
+        {"directory-image-exts",
+            OPT_STRINGLIST(directory_img_exts)},
         {0}
     },
     .size = sizeof(struct demux_playlist_opts),
     .defaults = &(const struct demux_playlist_opts){
         .dir_mode = DIR_AUTO,
+        .directory_filter = (char *[]){
+            NULL
+        },
+        .directory_vid_exts = (char *[]){
+            "3g2", "3gp", "avi", "flv", "m2ts", "m4v", "mj2", "mkv", "mov",
+            "mp4", "mpeg", "mpg", "ogv", "rmvb", "webm", "wmv", "y4m", NULL
+        },
+        .directory_aud_exts = (char *[]){
+            "aiff", "ape", "au", "flac", "m4a", "mka", "mp3", "oga", "ogg",
+            "ogm", "opus", "wav", "wma", NULL
+        },
+        .directory_img_exts = (char *[]){
+            "avif", "bmp", "gif", "j2k", "jp2", "jpeg", "jpg", "jxl", "png",
+            "svg", "tga", "tif", "tiff", "webp", NULL
+        },
```

Comment:
I think default extensions should be documented as well.

Reply:
Will think about it, generally `mpv --help=directory-video-exts` do the job, I think.